### PR TITLE
integrations: misc bug fixes

### PR DESCRIPTION
**Pull request by tynes**
## Description
Integration testing syncservice

## Contributing Agreement

- [x] I have read and understood the [Optimism Contributing Guide and Code of Conduct](./CONTRIBUTING.md) and am following those guidelines in this pull request.

### CODE REVIEW

File: rollup/sync_service.go [hunk 1]
```diff
@@ -15,12 +15,13 @@ import (
...
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/common/hexutil"
-	"github.com/ethereum/go-ethereum/contracts/addressresolver"
+	"github.com/ethereum/go-ethereum/contracts/addressmanager"
 	ctc "github.com/ethereum/go-ethereum/contracts/canonicaltransactionchain"
 	mgr "github.com/ethereum/go-ethereum/contracts/executionmanager"
 	"github.com/ethereum/go-ethereum/core"
 	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/ethereum/go-ethereum/ethdb"
+	"github.com/ethereum/go-ethereum/event"
```
Comment (tynes) [inline rollup/sync_service.go]: TODO: might be dead code

File: rollup/sync_service.go [hunk 2]
```diff
@@ -157,6 +160,8 @@ type SyncService struct {
 	clearTransactionsTicker          *time.Ticker
 	clearTransactionsAfter           uint64
 	heads                            chan *types.Header
+	chainAdds                        chan core.ChainEvent
+	chainFeed                        event.Subscription
```
Comment (tynes) [inline rollup/sync_service.go]: Dead code

File: rollup/sync_service.go [hunk 3]
```diff
@@ -165,15 +170,23 @@ type SyncService struct {
...
 	ctcDeployHeight                  *big.Int
 	AddressResolverAddress           common.Address
 	CanonicalTransactionChainAddress common.Address
-	L1ToL2TransactionQueueAddress    common.Address
 	SequencerDecompressionAddress    common.Address
 	StateCommitmentChainAddress      common.Address
 	ExecutionManagerAddress          common.Address
 }
 
+// subscribe to chain add event
```
Comment (tynes) [inline rollup/sync_service.go]: Delete this

File: rollup/sync_service.go [hunk 4]
```diff
@@ -423,26 +482,39 @@ func (s *SyncService) sequencerIngestQueue() {
...
+					rtx := txs[i]
+					log.Debug("Sequencer ingesting", "local-index", i, "rtx-index", rtx.index)
 					// set the timestamp
 					s.bc.SetCurrentTimestamp(rtx.timestamp.Unix())
 					// The god key needs to sign L1ToL2 transactions
 					err := s.maybeReorgAndApplyTx(rtx.index, rtx.tx, true)
 					if err != nil {
 						log.Error("Sequencer ingest queue transaction failed: %w", err)
 					}
+					// This executed value isn't being applied correctly
```
Comment (tynes) [inline rollup/sync_service.go]: dead comments

File: rollup/sync_service.go [hunk 5]
```diff
@@ -461,8 +533,9 @@ func (s *SyncService) sequencerIngestQueue() {
 				tip := s.bc.CurrentBlock()
 
 				isAtTip := tip.Number().Uint64() == totalElements.Uint64()
+				// Time stamp check isn't working
```
Comment (tynes) [inline rollup/sync_service.go]: TODO: delete the `ts` logic

File: rollup/sync_service.go [hunk 6]
```diff
@@ -571,17 +611,24 @@ func (s *SyncService) Stop() error {
 		defer s.cancel()
 	}
 
+	// TODO: this is potentially dead code
 	if s.headSubscription != nil {
```
Comment (tynes) [inline rollup/sync_service.go]: Delete dead code

File: rollup/sync_service.go [hunk 7]
```diff
@@ -571,17 +611,24 @@ func (s *SyncService) Stop() error {
...
 		defer s.headSubscription.Unsubscribe()
 	}
 
+	if s.chainFeed != nil {
+		defer s.chainFeed.Unsubscribe()
```
Comment (tynes) [inline rollup/sync_service.go]: Delete dead code

File: rollup/sync_service.go [hunk 8]
```diff
@@ -571,17 +611,24 @@ func (s *SyncService) Stop() error {
...
+
 	return nil
 }
 
 func (s *SyncService) Loop() {
+	log.Info("Starting Tip processing loop")
 	for {
 		select {
 		case header := <-s.heads:
+			log.Debug("Receiving header in main loop")
```
Comment (tynes) [inline rollup/sync_service.go]: delete this

File: rollup/sync_service.go [hunk 9]
```diff
@@ -699,8 +752,13 @@ func (s *SyncService) ProcessETHBlock(ctx context.Context, header *types.Header)
 	if header == nil {
 		return s.Eth1Data, errors.New("Cannot process nil header")
 	}
+	// defer catchPanic()
```
Comment (tynes) [inline rollup/sync_service.go]: Delete dead code or add panic handler

File: rollup/sync_service.go [hunk 10]
```diff
@@ -1054,3 +1133,14 @@ func (s *SyncService) applyTransaction(tx *types.Transaction) error {
 	}
 	return nil
 }
+
+// TODO(mark): just for debugging
+func prettyPrintTx(tx *types.Transaction, s string) {
```
Comment (tynes) [inline rollup/sync_service.go]: delete this

File: rollup/sync_service.go [hunk 11]
```diff
@@ -989,13 +1062,17 @@ func (s *SyncService) maybeReorgAndApplyTx(index uint64, tx *types.Transaction,
 // maybeApplyTransaction will look at the tips height and apply the transaction
 // if the transaction is at the next index. This allows the codepath to work for
 // the verifier as it syncs as well as the sequencer for reorgs.
+// This is currently subject to race conditions because the block production
+// goes through the miner. Cannot use.
 func (s *SyncService) maybeApplyTransaction(index uint64, tx *types.Transaction) error {
```
Comment (tynes) [inline rollup/sync_service.go:1035]: delete this

**Review by karlfloersch — APPROVED** [review]
Maybe rename to `Integrations: Misc bug fixes` or something?

Anyway, looking great!

**Pull request merged**